src/async-wrap.cc

```diff
@@ -296,7 +296,17 @@ static void PromiseHook(PromiseHookType type, Local<Promise> promise,
   PromiseWrap* wrap = Unwrap<PromiseWrap>(promise);
   if (type == PromiseHookType::kInit || wrap == nullptr) {
     bool silent = type != PromiseHookType::kInit;
-    wrap = new PromiseWrap(env, promise, silent);
+    // set parent promise's async Id as this promise's triggerId
+    if (parent->IsPromise()) {
+      // parent promise exists, current promise
+      // is a chained promise, so we set parent promise's id as
+      // current promise's triggerId
+      double trigger_id = Unwrap<PromiseWrap>(parent.As<Promise>())->get_id();
+      env->set_init_trigger_id(trigger_id);
+      wrap = new PromiseWrap(env, promise, silent);
```

Comment:
@AndreasMadsen , Thank very much for the review, I will try to understand this case and your solution.